sql-plugin/src/main/java/com/nvidia/spark/rapids/ColumnarCopyHelper.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids;
+
+import ai.rapids.cudf.HostColumnVector.ColumnBuilder;
+
+import org.apache.spark.sql.vectorized.ColumnVector;
+
+/**
+ * A helper class which efficiently transfers different types of host columnar data into cuDF.
+ */
+public class ColumnarCopyHelper {
+
+  public static void nullCopy(ColumnBuilder b, int rows) {
+    for (int i = 0; i < rows; i++) {
+      b.appendNull();
+    }
+  }
+
+  public static void booleanCopy(ColumnVector cv, ColumnBuilder b, int rows) {
+    if (!cv.hasNull()) {
+      for (int i = 0; i < rows; i++) b.append(cv.getBoolean(i));
+      return;
```

Comment:
nit: Is there any performance difference between using an early return like this and an else statement? If not I would prefer the else.

Reply:
I dumped JVM bytecode for both approaches. Compared with the else approach,  the return approach saves one `goto` instruction. And the `continue` approach is identical to the `else` approach in terms of bytecode:
The "return" approach:
```java
  public static void shortCopy(org.apache.spark.sql.vectorized.ColumnVector, ai.rapids.cudf.HostColumnVector$ColumnBuilder, int);
    Code:
       0: aload_0
       1: invokevirtual #3                  // Method org/apache/spark/sql/vectorized/ColumnVector.hasNull:()Z
       4: ifne          31
       7: iconst_0
       8: istore_3
       9: iload_3
      10: iload_2
      11: if_icmpge     30
      14: aload_1
      15: aload_0
      16: iload_3
      17: invokevirtual #9                  // Method org/apache/spark/sql/vectorized/ColumnVector.getShort:(I)S
      20: invokevirtual #10                 // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.append:(S)Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      23: pop
      24: iinc          3, 1
      27: goto          9
      30: return
      31: iconst_0
      32: istore_3
      33: iload_3
      34: iload_2
      35: if_icmpge     70
      38: aload_0
      39: iload_3
      40: invokevirtual #6                  // Method org/apache/spark/sql/vectorized/ColumnVector.isNullAt:(I)Z
      43: ifeq          54
      46: aload_1
      47: invokevirtual #2                  // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.appendNull:()Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      50: pop
      51: goto          64
      54: aload_1
      55: aload_0
      56: iload_3
      57: invokevirtual #9                  // Method org/apache/spark/sql/vectorized/ColumnVector.getShort:(I)S
      60: invokevirtual #10                 // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.append:(S)Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      63: pop
      64: iinc          3, 1
      67: goto          33
      70: return
```

The "else" approach:
```java
  public static void shortCopy(org.apache.spark.sql.vectorized.ColumnVector, ai.rapids.cudf.HostColumnVector$ColumnBuilder, int);
    Code:
       0: aload_0
       1: invokevirtual #3                  // Method org/apache/spark/sql/vectorized/ColumnVector.hasNull:()Z
       4: ifne          33
       7: iconst_0
       8: istore_3
       9: iload_3
      10: iload_2
      11: if_icmpge     30
      14: aload_1
      15: aload_0
      16: iload_3
      17: invokevirtual #9                  // Method org/apache/spark/sql/vectorized/ColumnVector.getShort:(I)S
      20: invokevirtual #10                 // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.append:(S)Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      23: pop
      24: iinc          3, 1
      27: goto          9
###       30: goto          72                     // the extra goto instruction of the "else" approach
      33: iconst_0
      34: istore_3
      35: iload_3
      36: iload_2
      37: if_icmpge     72
      40: aload_0
      41: iload_3
      42: invokevirtual #6                  // Method org/apache/spark/sql/vectorized/ColumnVector.isNullAt:(I)Z
      45: ifeq          56
      48: aload_1
      49: invokevirtual #2                  // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.appendNull:()Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      52: pop
      53: goto          66
      56: aload_1
      57: aload_0
      58: iload_3
      59: invokevirtual #9                  // Method org/apache/spark/sql/vectorized/ColumnVector.getShort:(I)S
      62: invokevirtual #10                 // Method ai/rapids/cudf/HostColumnVector$ColumnBuilder.append:(S)Lai/rapids/cudf/HostColumnVector$ColumnBuilder;
      65: pop
      66: iinc          3, 1
      69: goto          35
      72: return
```